### PR TITLE
Build release image

### PR DESCRIPTION
**Pull request by wangkuiyi**
This is part of https://github.com/sql-machine-learning/sqlflow/issues/2373.

We will build the step image later.

### CODE REVIEW

**Review by Yancey0623 — APPROVED** [review]
LGTM

File: docker/server/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+COPY build /build
+ENV SQLFLOW_PARSER_SERVER_PORT=12300
+ENV SQLFLOW_PARSER_SERVER_LOADING_PATH="/usr/local/sqlflow/java"
+
+# This image includes the SQLFlow gRPC server and external parser servers.  The
+# latter is in Java, so we need JRE.
+RUN apt-get -qq update \
+	&& apt-get -qq install -y openjdk-8-jre-headless \
+	&& mkdir -p "$SQLFLOW_PARSER_SERVER_LOADING_PATH" \
+	&& mv /build/*.jar "$SQLFLOW_PARSER_SERVER_LOADING_PATH" \
```
Comment (Yancey0623) [inline docker/server/Dockerfile]: We should  install the wheel package in server image:
``` bash
pip install --quiet /build/*.whl \
```

File: docker/server/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,21 @@
+# NOTE: The docker build context directory must be the root of the source tree.
+# NOTE: To build the release image, SQLFlow must be built into ./build
+FROM ubuntu:18.04
+
+# Install and configure the pre-built SQLFlow binaries.
+COPY build /build
+ENV SQLFLOW_PARSER_SERVER_PORT=12300
+ENV SQLFLOW_PARSER_SERVER_LOADING_PATH="/usr/local/sqlflow/java"
+
```
Comment (Yancey0623) [inline docker/server/Dockerfile]: `python/sqlflow_submitter` folder is also needed.

``` docker
COPY python /usr/local/sqlflow/python
ENV PYTHONPATH=/usr/local/sqlflow/python:$PYTHONPATH
```

**Review by Yancey0623 — CHANGES_REQUESTED** [review]

File: docker/server/Dockerfile (hunk 1 above)
Comment (wangkuiyi) [reply docker/server/Dockerfile]: I noticed that there are three wheel files in `./build`:

- couler*.whl
- fluid*.whl
- sqlflow_models*.whl

I think only the step image needs them, am I right?

File: docker/server/Dockerfile (hunk 2 above)
Comment (wangkuiyi) [reply docker/server/Dockerfile]: I assume these should be in the step image, other than the server image, am I right?

File: docker/server/Dockerfile (hunk 1 above)
Comment (lhw362950217) [reply docker/server/Dockerfile]: Server relies on couler to generate YAML file. So, we should install couler. We also need  to  install cubectl.

Comment (wangkuiyi) [reply docker/server/Dockerfile]: Oh. Yes! Thanks for the reminder!

File: docker/server/Dockerfile (hunk 2 above)
Comment (wangkuiyi) [reply docker/server/Dockerfile]: I moved them into the server image, as I think they are necessary for the server to all Fluid/Couler, right?

Comment (Yancey0623) [reply docker/server/Dockerfile]: Yes, that's right.

**Review by Yancey0623 — APPROVED** [review]
LGTM

**Pull request merged**